### ISSUE TITLE
Upgrade to Opam 2.0

### ISSUE THREAD

**Issue by CodeSteele** [AzuraCast/AzuraCast#955]
#876 pointed out we're on Opam 1.2 and was a major reason for the delay in us seeing a new version for duppy available to fix a bug with LiquidSoap.

Also the developer of LiquidSoap pointed out that further updates will only be available in the new Opam 2.0 package format, so we'll _have_ to update.

So best to see about doing that now.

**Comment by CodeSteele** [comment]
https://github.com/ocaml/opam/issues/3584

Looks like the Opam team is gunning for it to be available in 19.04, maybe we just wait on moving to 2.0 for now too?

**Comment by SlvrEagle23** [comment]
19.04 won't be an LTS release...for us it would mean waiting until 20.04. I'd like to find a PPA or some other more stable way of getting this done sooner if possible.

**Comment by SlvrEagle23** [comment]
Update: The `azuracast_radio` docker container has been updated to use a multistage build, separating out the Icecast-KH-AC compilation and the LS build into separate steps.

The good news is that this lets us take advantage of OPAM's own Docker images, and the total size of the final image is shrunk by a pretty remarkable 100MB, down from 430MB to about 330MB.

The bad news is that the build process for that image now takes a whopping 35 minutes, thanks in no small part to OPAM's insistence that if you want to add another directory as a "switch" (a virtual environment with a given compiler) it has to _compile OCAML from scratch_ inside that directory. Sheesh.

**Comment by CodeSteele** [comment]
Isn't that going to _kill_ us in development?

**Comment by SlvrEagle23** [comment]
@CodeSteele Unlike the Docker Hub, local docker builds take advantage of intermediate image caches. They also run way faster in general in the first place. Most rebuilds won't be nearly that bad on local dev.

**Comment by CodeSteele** [comment]
Yeah, I guess with the split images it's not that big a deal -- prior we built Icecast before Opam ran so we'd be waiting 35+ minutes on every Icecast rebuild.

Buttttttt yeah that's no longer the case so I guess no big deal.

**Comment by SlvrEagle23** [comment]
With LS + Icecast: 40+ minutes

With Icecast: 20 minutes

Without either: 18 minutes

I'm not particularly happy with _any_ of those build times. I will look into spinning up Drone again and getting it to publish to the Docker hub, as it will properly use Docker intermediate image caching and can thus build/push minor changes MUCH faster.
